lte/protos/session_manager.proto

```diff
@@ -540,3 +542,255 @@ service CentralSessionController {
+
+
+//RequestedQosRules
+message RequestedQosRules {
+   uint32 qos_rule_identifier = 1;
+   bool dqr = 2;
+   uint32 number_of_packet_filters = 3;
+   repeated int32 packet_filter_identifier = 4;
+   uint32 qos_rule_precedence = 5;
+   bool segregation = 6;
+   uint32 qos_flow_identifier = 7;
+}
+
+//RequestedQosFlowDescriptions
+//message AuthorizedQosFlowDescriptions {
+  //  uint32 qfi = 0;
+//  QosFlowOperationCode operation_code = 1;     //Create or modify or delete
+  //  bool e_bit = 2;
+  //  uint32 number_of_parameters = 3;
+  //  repeated ParameterIdentifier parameter_identifier = 4;
+  //  repeated int32 length_of_parameter_contents = 5;
+  //  repeated string parameter_contents = 6;
+//}
+//enum QosFlowOperationCode {
+  //  CREATE_NEW_QOS_FLOW_DESCRIPTION = 0;
+  //  DELETE_EXISTING_QOS_FLOW_DESCRIPTION = 1;
+  //  MODIFY_EXISTING_QOS_FLOW_DESCRIPTION = 2;
+//}
+//enum ParameterIdentifier {
+//    5QI = 0;
+//    GFBR_UPLINK = 1;
+//    GFBR_DOWNLINK = 2;
+//    MFBR_UPLINK = 3;
+
+//    MFBR_DOWNLINK = 4;
+//    AVERAGING_WINDOW = 5;
+//    EPS_BEARER_IDENTITY = 6;
+//}
+
+//mapped_eps_bearer_context
+//message MappedEpsBearerContext {
+//   uint32 eps_bearer_identity = 0; //24.301 - 9.4.2 EPS bearer identity from the value range 5 to 15.
+//   BearerOperationCode operation_code = 1;
+//   EBit e_bit = 2;
+//   repeated EpsParameterIdentity eps_parameter_identifier = 3;
+//   repeated int32 length_of_the_EPS_parameter_contents = 4;
+//   repeated string eps_parameter_contents = 5;
+//}
+//enum BearerOperationCode {
+//   CREATE_NEW_EPS_BEARER = 0;
+//   DELETE_EXISTING_EPS_BEARER = 1;
+//   MODIFY_EXISTING_EPS_BEARER = 2;
+//}
+//enum EpsParameterIdentity {
+//    MAPPED_EPS_QOS_PARAMETERS = 0;
+//    MAPPED_EXTENDED_EPS_QOS_PARAMETERS =1;
+//    TRAFFIC_FLOW_TEMPLATE = 2;
+//    APN_AMBR = 3;
+//    EXTENDED_APN_AMBR = 4;
+//}
+//ExtProtocolConfigOption
+//message ExtProtocolConfigOption {
+//   repeated uint32 protocolid = 0;
+//   repeated int32 length_of_protocol_content = 1;
+//   repeated string protocol_content = 2;
+//   repeated uint32 containerid = 3;
+//   repeated int32 length_of_container_content = 4;
+//   repeated string container_content = 5;
+//}
+//LadnServiceArea
+
+message LadnServiceArea {
+   string dnn_value = 1;
+   repeated TrackingAreaIdentityList identity_list = 2;
+}
+message TrackingAreaIdentityList {
+   repeated TypeOfList type_of_list = 1;
+   repeated int32 number_of_elements = 2;
+   string MCC=3;
+   string MNC=4;
+   string TAC= 5;          // tac is havaing length 8 bytes,MCC is 3 bytes, MNC is 3 bytes
+}
+enum TypeOfList  {
+   LIST_OF_TACS_BELONGING_TO_ONE_PLMN_WITH_NON_CONSECUTIVE_TAC_VALUES = 0;
+   LIST_OF_TACS_BELONGING_TO_ONE_PLMN_WITH_CONSECUTIVE_TAC_VALUES = 1;
+   LIST_OF_TAIS_BELONGING_TO_DIFFERENT_PLMNS = 2;
+}
+enum priorityaccess  {
+   High = 0;
+   Normal = 1;
+   Low = 2;
+}
+
+enum SMSessionFSMState {
```

Comment:
We could use this enum directly in SessionState, might save an extra step of translating

Reply:
In SessionD we have used enum string as CTEATING, CREATED etc. While using same string its, conflicting, hence named as CRTEATING_0, CREATED_1. After complete refactor, we can use directly this FSMState